### PR TITLE
OpenCensus metric exporter bridge

### LINKED ISSUE #93: Shim: OpenCensus Shim
Create an OpenCensus shim that bridges opencensus to opentelemetry.

### PR DESCRIPTION
**Pull request by dashpole**
Fixes #93

This allows wrapping an OpenTelemetry exporter so that it can be used with OpenCensus.  This is useful to facilitate migrations from OpenCensus to OpenTelemetry, as it would allow using a mix of OpenCensus and OpenTelemetry APIs in code (and libraries), while sending all telemetry to a single exporter.

Everything was straightforward to map, except the data points themselves.  To do that, I ended up having to implement custom `Aggregation`s based on the data point type.

I also need to do an audit of OpenCensus functionality that is "dropped" when using the bridge and document it.  Thus far, I know it drops `Summary` data points, exemplars, and some types of metric descriptors.  I might do that in a follow-up.

### CODE REVIEW

File: bridge/opencensus/aggregation.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+// aggregationWithEndTime is an aggregation that can also provide the timestamp
+// of the last recorded point.
+type aggregationWithEndTime interface {
+	aggregation.Aggregation
+	end() time.Time
+}
+
+// newAggregationFromPoints creates an OpenTelemetry aggregation from
+// OpenCensus points.  Points may not be empty.
```
Comment (Aneurysm9) [inline bridge/opencensus/aggregation.go:42]:
```suggestion
// OpenCensus points.  Points may not be empty and must be either
// all (int|float)64 or all *metricdata.Distribution.
```

File: bridge/opencensus/aggregation.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+			points[i] = number.NewInt64Number(pt.Value.(int64))
+		case float64:
+			points[i] = number.NewFloat64Number(pt.Value.(float64))
+		default:
+			return nil, fmt.Errorf("%w: %v", errIncompatibleType, t)
+		}
+	}
+	return &ocExactAggregator{
+		points:  points,
+		endTime: pts[len(pts)-1].Time,
```
Comment (Aneurysm9) [inline bridge/opencensus/aggregation.go:80]: Are all values in `points` guaranteed to have the same ending time or be monotonically increasing with respect to time?  Should this assumption be documented?

**Review by Aneurysm9 — APPROVED** [review]

File: bridge/opencensus/aggregation.go (hunk 2 above)
Comment (dashpole) [reply bridge/opencensus/aggregation.go:81]: Looking at where the list of points is constructed: [raw metric src](https://github.com/census-instrumentation/opencensus-go/blob/master/metric/common.go#L86), [view src](https://github.com/census-instrumentation/opencensus-go/blob/master/stats/view/view_to_metric.go#L124), there is only ever a single point in the list of points.  I'll document it.

**Comment by dashpole** [comment]
Friendly ping for reviews!

File: bridge/opencensus/README.md [hunk 3]
```diff
@@ -63,10 +67,56 @@ octrace.DefaultTracer = opencensus.NewTracer(tracer)
...
+OpenTelemetry pipeline so that migrating a metric does not require maintaining
+separate export pipelines for OpenCensus and OpenTelemetry.
+
+### The Exporter "wrapper" solution
+
+The solution we use here is to allow wrapping an OpenTelemetry exporter such
+that it implements the OpenCensus exporter interfaces. This allows a single
+exporter to be used for metrics from *both* OpenCensus and OpenTelemetry.
+
+### User Journy
```
Comment (sirianni) [inline bridge/opencensus/README.md:95]:
```suggestion
### User Joureny
```

**Comment by dashpole** [comment]
rebased again.  @jmacd I think you said you wanted to take a look at this.

File: bridge/opencensus/README.md (hunk 3 above)
Comment (MrAlias) [reply bridge/opencensus/README.md:95]: This doesn't seem to have been resolved.

Comment (dashpole) [reply bridge/opencensus/README.md]: applied

**Review by MrAlias — APPROVED** [review]

**Pull request merged**